google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -1330,6 +1331,166 @@ private Map<String, Object> getData() throws ExecutionException, InterruptedExce
+                          break;
+                        case 1:
+                          assertNotNull(docSnap);
+
+                          assertEquals(
+                              docSnap.getVectorValue("vector0"),
+                              FieldValue.vector(new double[] {0.0}));
+                          assertEquals(
+                              docSnap.getVectorValue("vector1"),
+                              FieldValue.vector(new double[] {1, 2, 3.99}));
+
+                          ref.set(
+                              map(
+                                  "purpose",
+                                  "vector tests",
+                                  "vector0",
+                                  FieldValue.vector(new double[] {0.0}),
+                                  "vector1",
+                                  FieldValue.vector(new double[] {1, 2, 3.99}),
+                                  "vector2",
+                                  FieldValue.vector(new double[] {0, 0, 0})));
+                          break;
+                        case 2:
+                          assertNotNull(docSnap);
+
+                          assertEquals(
+                              docSnap.getVectorValue("vector0"),
+                              FieldValue.vector(new double[] {0.0}));
+                          assertEquals(
+                              docSnap.getVectorValue("vector1"),
+                              FieldValue.vector(new double[] {1, 2, 3.99}));
+                          assertEquals(
+                              docSnap.getVectorValue("vector2"),
+                              FieldValue.vector(new double[] {0, 0, 0}));
+
+                          ref.update(
+                              map("vector3", FieldValue.vector(new double[] {-1, -200, -999})));
+                          break;
+                        case 3:
+                          assertNotNull(docSnap);
+
+                          assertEquals(
+                              docSnap.getVectorValue("vector0"),
+                              FieldValue.vector(new double[] {0.0}));
+                          assertEquals(
+                              docSnap.getVectorValue("vector1"),
+                              FieldValue.vector(new double[] {1, 2, 3.99}));
+                          assertEquals(
+                              docSnap.getVectorValue("vector2"),
+                              FieldValue.vector(new double[] {0, 0, 0}));
+                          assertEquals(
+                              docSnap.getVectorValue("vector3"),
+                              FieldValue.vector(new double[] {-1, -200, -999}));
+
+                          ref.delete();
+                          break;
+                        case 4:
+                          assertNull(docSnap);
+                          break;
+                      }
+                    } catch (AssertionError e) {
```

Comment:
I think you should expand this to all Exceptions, so that you don't swallow them by accident and let the test pass.

Reply:
TY. Expanded to catch `Throwable`.